app/models/concerns/form526_rapid_ready_for_decision_concern.rb

```diff
@@ -109,9 +109,9 @@ def open_claims
   # fetch, memoize, and return all of the veteran's rated disabilities from EVSS
   def all_rated_disabilities
     settings = Settings.lighthouse.veteran_verification.form526
+    icn = UserAccount.where(id: user_account_id).first&.icn
```

Comment:
If you're searching by `id`, I think you can just search with .find. So this could turn into,
```
icn = UserAccount.find(user_account_id)&.icn
```
non-blocking comment, as I realize using `where` might have other benefits here that I'm not aware of.

Reply:
@navaeddie 
`user_account_id` is on the `Form526Submission` class.
(you can check in the database table `form526_submissions`)
this concern is included in that class and it is what this method is called in context of - a `Form526Submission`.

Reply:
@navaeddie 
re: `.find` vs. `.where`

this is more of a question of "when do you want the application to error out?"

if we use `.find` here and an account is not found, the application will error out immediately, which makes the safe navigator useless, haha

if we use `.where` and an account is not found, the application will continue until it reaches the API call, which may tell us there is no ICN found and it will send the error to sentry, at least with the way we have it set up now.

there may be a bigger story here of my first question (when should we error out?) but i think that is handled by our ServiceException class for now without tearing the existing code up.

i suggest we make a lower-priority ticket to see how we can handle errors like this in the job context.

what do you think?

Reply:
and the `pluck` i don't know about.

you'd get `id` and `icn` in an array of arrays, something like `[[id, icn, time, time]]``with that query, so, that may be a little gnarly to parse out. 

however, i am indifferent :shrug:  hehe

i think the better way to do it would be a `select` statement (`UserAccount.select(:icn).where(id: user_account_id')`, but i wouldn't know how to benchmark that, haha

i'll leave that up to seth :rofl:

Reply:
> i think the better way to do it would be a `select` statement (`UserAccount.select(:icn).where(id: user_account_id)`, but i wouldn't know how to benchmark that, haha
> 
> i'll leave that up to seth 🤣

I did give benchmarking the old college try, but it did not bear fruit. Locally, the rails console does give me query execution times, but there seems to be a query cache in play that I just can't seem to get around. Rails consoles in higher environments aren't even configured to display execution times. In any case, I don't get the sense that either way is significantly faster than the other- so I'm inclined to keep it all as-is for now

also, the lower-priority error-handling ticket sounds like a good idea, I'll create this unless @navaeddie thinks otherwise...

Reply:
Created the error handling [follow up story](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/58513) @ahay-agile6 @navaeddie